client/components/layout/Navigation.tsx

```diff
@@ -0,0 +1,78 @@
+import { Box, Button } from '@mui/material';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import React from 'react';
+
+import { useAuthDispatch, useAuthState } from 'client/contexts/auth';
+import createAxiosInstance from 'client/lib/axios';
+import { getDiscordOauthUrl } from 'client/lib/oauth';
+
+interface NavLink {
+  label: string;
+  href?: string;
+  onClick?: () => void;
+}
+
+const Navigation: React.FC = () => {
+  const { authed } = useAuthState();
+  const authDispatch = useAuthDispatch();
+  const router = useRouter();
+
+  const DIRECTORY_LINK: NavLink = {
+    label: 'Discover',
+    href: '/directory',
+  };
+
+  const PROFILE_LINK: NavLink = {
+    label: 'Profile',
+    href: '/profile',
+  };
+
+  const LOGIN_LINK: NavLink = {
+    label: 'Login',
+    href: getDiscordOauthUrl(),
+  };
+
+  const LOGOUT_LINK: NavLink = {
+    label: 'Logout',
+    onClick: async () => {
+      const axios = createAxiosInstance();
+      await axios.post('/api/auth/logout');
+      authDispatch({ type: 'SET_AUTHED', authed: false });
```

Comment:
it won't redirect on a failed login; the promise will reject - kind of punting error handling as a whole since we don't really do it in the app; we should get that up and running once we have [stackable snacks](https://mui.com/material-ui/react-snackbar/#notistack)